src/app/shared/services/topbar.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { WalletService } from './wallet/wallet.service';
+import { Subject, Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TopbarService {
+  private subject = new Subject<any>();
+
+  constructor(private walletService: WalletService) {}
+
+  getWalletState(): Observable<any> {
+    this.walletService.getUserWallet().subscribe(res => {
+      this.subject.next(true);
+    }, err => {
+      this.subject.next(false);
+  });
+    return this.subject.asObservable();
+  }
+
+  walletInInitProcess(isWalletInit: boolean) {
+    this.subject.next(isWalletInit);
+  }
```

Comment:
You missed the point about observables and subject here. They shouldn't be used like that. You made nothing with using subject here. You could do it without using it. We'll discuss about it later.

You don't need TopbarService however you think about it. The only thing that you need in reminders component is to subscribe to change observables inside wallet and user service. that's the only thing that it needs.